helm-app-operator/pkg/helm/engine.go

```diff
@@ -49,24 +54,40 @@ func (o *OwnerRefEngine) Render(chart *chart.Chart, values chartutil.Values) (ma
 }
 
 // addOwnerRefs adds the configured ownerRefs to a single rendered file
+// Adds the ownerrefs to all the documents in a YAML file
```

Comment:
Can we just use the `SplitManifests` util from helm so we don't have to re-implement here?

https://github.com/helm/helm/blob/7cad59091a9451b2aa4f95aa882ea27e6b195f98/pkg/releaseutil/manifest.go#L38

Reply:
```
manifests := releaseutil.SplitManifests(fileContents)
for _, manifest := range manifests {
	manifestMap := chartutil.FromYaml(manifest)
	if errors, ok := manifestMap["Error"]; ok {
		return "", fmt.Errorf("error parsing rendered template to add ownerrefs: %v", errors)
	}
	unst, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&manifestMap)
	if err != nil {
		return "", err
	}
	unstructured := &unstructured.Unstructured{Object: unst}
	unstructured.SetOwnerReferences(o.refs)
	err = encoder.Encode(unstructured.Object)
	if err != nil {
		return "", fmt.Errorf("error parsing the object to yaml: %v", err)
	}
}
```

Does it Looks fine ?. I think the `encoder` part is still required to combine the manifests at the end unless if we go for string append ?

Reply:
@ecordell Any updates ?